aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+    return [t[1] for t in tuples]
+
+
+def check_file(nc, site_code):
+    """
+    Return list of errors found in the file:
+    Variables of interest are present
+    TIME. LATITUDE, LONGITUDE,  is present
+    NOMINAL_DEPTH is not present as variable or attribute
+    file_version is not FV01
+    if LATITUDE or LONIGUTDE dimension has length >1
+
+    :param file: name of the netcdf file
+    :param nc: xarray dataset
+    :param VoI: string. Variable of Interest
+    :param site_code: code of the mooring site
+    :return: dictionary with the file name and list of failed tests
+    """
+
+    attributes = list(nc.attrs)
+    file_variables = list(nc.variables)
+    allowed_dimensions = ['TIME', 'LATITUDE', 'LONGITUDE', 'HEIGHT_ABOVE_SENSOR']
+    required_variables = ['UCUR', 'VCUR', 'WCUR']
+    error_list = []
+
+    nc_site_code = nc.site_code
+    if nc_site_code != site_code:
+        error_list.append('Wrong site_code: ' + nc_site_code)
+
+    nc_file_version = nc.file_version
+    if 'Level 1' not in nc_file_version:
+        error_list.append('Wrong file version: ' + nc_file_version)
+
+    for variable in required_variables:
+        if variable not in file_variables:
+            error_list.append(variable + ' variable missing')
+        else:
+            VoIdimensions = list(nc[variable].dims)
+            if 'TIME' not in VoIdimensions:
+                error_list.append('TIME is not a dimension for ' + variable)
+            if 'LATITUDE' in VoIdimensions and len(nc.LATITUDE) > 1:
+                error_list.append('more than one LATITUDE for ' + variable)
+            if 'LONGITUDE' in VoIdimensions and len(nc.LONGITUDE) > 1:
+                error_list.append('more than one LONGITUDE for ' + variable)
+            for dim in VoIdimensions:
+                if dim not in allowed_dimensions:
+                    error_list.append('not allowed dimension: ' + dim)
+
+    if 'NOMINAL_DEPTH' not in file_variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
+
+    return error_list
+
+
+def get_nvalues(nc):
+    """
+    Get the number of cells above the sensor
+    :param nc: xarray dataset
+    :return: number of cells, number of bins
+    """
+    if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+        nbins = nc.dims['HEIGHT_ABOVE_SENSOR']
+        nvalues = nc.dims['TIME'] * nbins
+    else:
+        nbins = 1
+        nvalues = nc.dims['TIME']
+    return nvalues, nbins
+
+
+def get_varvalues(nc, varname):
+    """
+    Return a 1D array of 2D values
+    :param nc: dataset
+    :param varname: Variable of interest
+    :return: variable values flattened
+    """
+    return nc[varname].values.flatten()
+
+
+def get_instrumentID(nc):
+    """
+    Create instrument id based on deployment metadata
+    :param nc: xarray dataset
+    :return: instrumentID as string
+    """
+    return '; '.join([nc.deployment_code, nc.instrument, nc.instrument_serial_number])
+
+
+def in_water(nc):
```

Comment:
You don't need to use the "nc" class as argument, you just need an array and two strings.
Hence, If it defined as ```def get_in_water(time, start_str, end_str)```,
the code call would be:

```
time = nc['TIME']
in_water_ind = get_in_water(time,nc['time_coverage_start'],nc['time_coverage_end'])
```

Again, intent is clear at the function call.

Reply:
I just noticed that the variable name here is confusing - you use ```nc``` to reference to a ```xarray``` dataset. Just rename the variable to ```xrobj ``` or ```xobj```. I did the comment above expecting the arguments to be netCDF4 objects.

This is a case where type hints shines.

Reply:
nc for xarray and ds to netcdf4 dataset